### ISSUE TITLE
Make static adapter truly static

### ISSUE THREAD

**Issue by lsabi** [sveltejs/kit#1453]
**Is your feature request related to a problem? Please describe.**
I'm not sure if it's a missing feature, an intended "feature" or just a bug.

Basically, I have a website with some dynamic pages (slugs) that I know at compile time. I want to generate the static pages and make them accessible via a traditional hosting (e.g. apache webserver).

The use case is a blog, where I have dynamic slugs that are fetched at build time and used to generate static pages. The problem is, though, the pages are not generated, instead are left dynamic when using the `static adapter` and thus, navigating through the web pages works (starting from the root), but when accessing the page directly, this is not found.

**Describe the solution you'd like**
To be able to generate true static pages, creating an `html` file for all routes, even the dynamic ones. If the target is to create a static website, then it should be possible to have all the pages rendered and stored independently one from another.

Basically, pages like `/blog/[slug].svelte` should be translated to `/blog/my-post-name.html` instead of being dynamic even when using the static adapter.

I can't recall if it was working before the last update of the static adapter or not. I believe it was

**Describe alternatives you've considered**
I don't have other alternatives at the moment.

**How important is this feature to you?**
It's quite serious, since a user who visits a dynamic page and refreshes the page by accident, gets a `404 - Not found` from the apache webserver.

**Additional context**
Thanks

**Comment by babichjacob** [comment]
This is already how it works?

**Comment by lsabi** [comment]
No, it is not. With `sapper` it worked this way, and all the dynamic pages were generated (dynamic folders as well). I'm using the default configuration in `svelte.config.js`

**Comment by Conduitry** [comment]
Please provide a repro. Links in HTML should already be getting followed when crawling like in Sapper.

**Comment by lsabi** [comment]
@Conduitry the project I'm experiencing the error is from a migration from Sapper to Kit. I have another Kit project with the same config, it's just some versions behind (it uses `svelte.config.cjs`) and both static and dynamic pages/folders are generated.

Since the project I'm working on is private, I wanted to reproduce it on a fresh new repo, though the static adapter seems not to be working (at least, not as in the other project which I updated last night, while the new one today morning).

Here's the repo [https://github.com/lsabi/kit-static-error](https://github.com/lsabi/kit-static-error)

Unfortunately, the docs for the adapter are quite bare, so it's difficult to tell if in this case it's my error or not.

**Comment by DryCreations** [comment]
looks like you need to add a favicon to stop the error during build (or take the line out that's requesting a favicon).

l also had to run `npm install @sveltejs/adapter-static@next` before it would work.

i'd suggest the following to fix your issue:

1. delete `/build`
2. `npm install @sveltejs/adapter-static@next`
3. add `/static/favicon.ico`
4. `npm run build`

additionally, if you don't want to rely on the crawler to find pages, or if the dynamic routes you want aren't being generated, you can try adding them explicitly in the config:

```js
kit: {
  adapter: adapter({out: "build"}),
  target: "#svelte",
  prerender: {
    pages: ['*', 'example/stuff2']
  },
}
```

I forked your repo and got it into a [working state](https://github.com/DryCreations/kit-static-error), it's also being deployed to gh-pages for proof. you can see the generated files [here](https://github.com/DryCreations/kit-static-error/tree/gh-pages), or the deployed site [here](https://drycreations.github.io/kit-static-error/)

I had to make a couple additional changes so that it can handle the base url for github pages. it should work for you without those changes.

**Comment by lsabi** [comment]
@DryCreations thanks for looking into it.

I followed the tutorial on the migrations page and I stepped into this problem, which after the fixes you suggested, is still giving me the same error. Though, if I explicitly set the pages to be exported, aside from "*", it exports the pages I've set. Unfortunately with dynamic routes it is not simple to define all the routes in the `svelte.config.js` file especially since I have many of those.

I believe the problem could lie in the fact that I also have a folder `[category]` with an `index.svelte` file in it. If I print the `category` from the `page.params` within the `load` function, I get `service-worker.js`. Could that raise a not found error and interfere with the build process?

Regarding the creation of the `favicon.ico`, an error could be thrown instead of a silent one. I wasn't aware of the fact that it is mandatory and I've never thought of it for a second.
